### PR TITLE
Kubenetes add missing environment variables

### PR DESCRIPTION
**Pull request by longshuicy**
- deployment.yaml updated to use secrets and value variables properly
- pushed the value to box folder "chen-clowder2-software-dev.yaml" ( i can also overwrite the default clowder2-software-dev.yaml but I was unsure)
site up running at https://clowder2.software-dev.ncsa.cloud/ please test out

### CODE REVIEW

File: charts/clowder2/templates/backend/deployment.yaml [hunk 1]
```diff
@@ -62,6 +73,22 @@ spec:
               value: $(CLOWDER2_URL)
             - name: elasticsearch_url
               value: http://{{ include "clowder2.name" . }}-elasticsearch:9200
+            - name: elasticsearch_no_of_shards
+              value: 5
+            - name: elasticsearch_no_of_replicas
+              value: 5
+#              TODO use telegraf.sealed.yaml
+            - name: RABBITMQ_USER
+              value: guest
```
Comment (robkooper) [inline charts/clowder2/templates/backend/deployment.yaml:82]: this should not be hardcoded

File: charts/clowder2/templates/backend/deployment.yaml [hunk 2]
```diff
@@ -62,6 +73,22 @@ spec:
...
+            - name: RABBITMQ_PASS
```
Comment (robkooper) [inline charts/clowder2/templates/backend/deployment.yaml:83]: this should not be hardcoded

File: charts/clowder2/templates/backend/deployment.yaml [hunk 3]
```diff
@@ -40,8 +40,19 @@ spec:
               value: "4"
             - name: MINIO_SERVER_URL
               value: {{ include "clowder2.name" . }}-minio-headless:9000
+            - name: MINIO_BUCKET_NAME
+              value: clowder
+#              TODO use telegraf.sealed.yaml
+            - name: MINIO_ACCESS_KEY
+              value: minioadmin
```
Comment (robkooper) [inline charts/clowder2/templates/backend/deployment.yaml:47]: this should not be hardcoded

File: charts/clowder2/templates/backend/deployment.yaml [hunk 4]
```diff
@@ -40,8 +40,19 @@ spec:
...
+            - name: MINIO_SECRET_KEY
+              value: minioadmin
```
Comment (robkooper) [inline charts/clowder2/templates/backend/deployment.yaml:49]: this should not be hardcoded

File: charts/clowder2/templates/backend/deployment.yaml [hunk 5]
```diff
@@ -40,8 +40,19 @@ spec:
               value: "4"
             - name: MINIO_SERVER_URL
               value: {{ include "clowder2.name" . }}-minio-headless:9000
+            - name: MINIO_BUCKET_NAME
+              value: clowder
```
Comment (robkooper) [inline charts/clowder2/templates/backend/deployment.yaml:44]: this should not be hardcoded

File: charts/clowder2/templates/backend/deployment.yaml [hunk 6]
```diff
@@ -40,8 +40,19 @@ spec:
...
+#              TODO use telegraf.sealed.yaml
+            - name: MINIO_ACCESS_KEY
+              value: minioadmin
+            - name: MINIO_SECRET_KEY
+              value: minioadmin
+            - name: MINIO_UPLOAD_CHUNK_SIZE
+              value: 10 * 1024 * 1024
```
Comment (robkooper) [inline charts/clowder2/templates/backend/deployment.yaml:51]: can we set this in values.yaml?

**Review by robkooper — CHANGES_REQUESTED** [review]

File: scripts/keycloak/full-kube-clowder-realm-prod.json [hunk 7]
```diff
@@ -0,0 +1,1914 @@
+{
```
Comment (longshuicy) [inline scripts/keycloak/full-kube-clowder-realm-prod.json:1]: Maybe just keep it here for now until we figure out a more generic way to set this. i.e. pass it in as a file to the helm chart

**Comment by longshuicy** [comment]
Make sure we limit the Access to IPs within NCSA or UIUC

**Review by lmarini — APPROVED** [review]

**Pull request merged**